common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -4,50 +4,267 @@
+        scope=Scope.settings,
+        default=0)
+    weight = Float(
+        display_name=_("Problem Weight"),
+        help=_("Defines the number of points each problem is worth. "
+               "If the value is not set, each response field in the problem is worth one point."),
+        values={"min": 0, "step": .1},
+        scope=Scope.settings
+    )
+    markdown = String(help=_("Markdown source of this module"), default=None, scope=Scope.settings)
+    source_code = String(
+        help=_("Source code for LaTeX and Word problems. This feature is not well-supported."),
+        scope=Scope.settings
+    )
+    use_latex_compiler = Boolean(
+        help=_("Enable LaTeX templates?"),
+        default=False,
+        scope=Scope.settings
+    )
+    matlab_api_key = String(
+        display_name=_("Matlab API key"),
+        help=_("Enter the API key provided by MathWorks for accessing the MATLAB Hosted Service. "
+               "This key is granted for exclusive use by this course for the specified duration. "
+               "Please do not share the API key with other courses and notify MathWorks immediately "
+               "if you believe the key is exposed or compromised. To obtain a key for your course, "
+               "or to report an issue, please contact moocsupport@mathworks.com"),
+        scope=Scope.settings
+    )
+
+    INDEX_CONTENT_TYPE = 'CAPA'
+
+    resources_dir = None
+
+    has_score = True
+    show_in_read_only_mode = True
+    template_dir_name = 'problem'
+    mako_template = "widgets/problem-edit.html"
+    has_author_view = True
+
+    # The capa format specifies that what we call max_attempts in the code
+    # is the attribute `attempts`. This will do that conversion
+    metadata_translations = dict(XmlParserMixin.metadata_translations)
+    metadata_translations['attempts'] = 'max_attempts'
+
     icon_class = 'problem'
 
-    js = {
+    uses_xmodule_styles_setup = True
+
+    preview_view_js = {
         'js': [
             resource_string(__name__, 'js/src/javascript_loader.js'),
             resource_string(__name__, 'js/src/capa/display.js'),
             resource_string(__name__, 'js/src/collapsible.js'),
             resource_string(__name__, 'js/src/capa/imageinput.js'),
             resource_string(__name__, 'js/src/capa/schematic.js'),
+        ],
+        'xmodule_js': resource_string(__name__, 'js/src/xmodule.js')
+    }
+
+    preview_view_css = {
+        'scss': [
+            resource_string(__name__, 'css/capa/display.scss'),
+        ],
+    }
+
+    studio_view_js = {
+        'js': [
+            resource_string(__name__, 'js/src/problem/edit.js'),
+        ],
+        'xmodule_js': resource_string(__name__, 'js/src/xmodule.js'),
+    }
+
+    studio_view_css = {
+        'scss': [
+            resource_string(__name__, 'css/editor/edit.scss'),
+            resource_string(__name__, 'css/problem/edit.scss'),
         ]
     }
 
-    js_module_name = "Problem"
```

Comment:
This is used by `shim_xmodule_js()`. Have changed `shim_xmodule_js()` to accept the module name as a parameter and passed in from `student_view()`.